### DISCUSSION TITLE
Need to import additional Books/Sermons/Dictionaries to Read/View

### DISCUSSION THREAD

**Discussion by passionate2023** [ezra-bible-app/ezra-bible-app#910] [Ideas]
Can you please provide import options to import for additional books/sermons/dictionaries to read/view along with the bible?

a. sample file format (.txt, .json)
b. import file option to import

**Comment by tobias-klein** [comment]
The way to achieve this is to convert existing sources to sword modules and installing them in your local sword folder. The purpose of Ezra Bible App is not to be a converter/import tool for text sources. There are other tools for that. Therefore, I don't see how this will be implemented in this app. However, please consider the possibility of getting in touch with the SWORD community to create sword modules from any available free sources.

See https://www.crosswire.org/sword/develop/swordmodule/

**Comment by passionate2023** [comment]
If i create a Sword module, how to import the sword module into Ezra Bible App. Is it possible to import the Sword module into Ezra app?

Does the sword module works with the sword bible software or both (sword and ezra) ?

**Reply by tobias-klein** [reply]
Ezra Bible App is a SWORD frontend like various other frontends (see list [here](https://crosswire.org/applications/)).

This means the approach for installing additional modules is the same as with other SWORD frontends.

On Windows the path to the SWORD folder where it holds all its modules is:
C:\Users\<username>\AppData\Roaming\sword

Within that folder you find two more:
* mods.d => Contains all config files for installed modules.
* modules => Contains the actual module data.

Currently the app supports Bible modules, Dictionary modules and with the upcoming latest release also Commentary modules.

Have a look here for some details regarding module development:
https://wiki.crosswire.org/DevTools:Modules
